### ISSUE TITLE
Mario cannot rotate right properly on trees and poles

### ISSUE THREAD

**Issue by Sporkinator** [sm64pc/sm64pc#204]
**Describe the bug**
Mario can rotate left, but not right.

**To Reproduce**
Climb a tree or a pole. Hold the analog stick to the left. Mario rotates indefinitely. This is fine. Now, hold the analog stick to the right. Mario will rotate for a bit and then stop for no reason. He is unable to rotate right indefinitely.

**Expected behavior**
In the N64 version, the rotation is indefinite in both directions. This is the expected behavior.

**Desktop (please complete the following information):**
 - OS: Windows 7 Pro x64, running OpenGL build.

**Comment by dgeneratioN** [comment]
I don't think this is only related to poles, as turning while standing still or walking is weird compared to the n64 version (on the default camera build atleast)

**Comment by fgsfdsfgs** [comment]
I can't reproduce this neither on master nor on nightly. What build are you using? What flags was it built with?

**Comment by porcino** [comment]
This bug is present in every build for me, even in one from may 4th, no matter the build flags, same OS (w7 x64)

**Comment by dgeneratioN** [comment]
Was built with no flags and NODRAWDISTANCE=1, using xbox one controller on W10

**Comment by uper333** [comment]
It happens to me too compiled with BETTERCAMERA=1 NODRAWINGDISTANCE=1

**Comment by GateGuy** [comment]
It also happens to me on Windows 10 x64 with NODRAWINGDISTANCE=1

**Comment by fgsfdsfgs** [comment]
I still can't reproduce this even on the latest nightly. Windows 7 x64, `BETTERCAMERA=1`, `NODRAWINGDISTANCE=1` or `0`, it doesn't matter, x86_64 MinGW-W64 build, XBOne controller. Only tried on trees, but should be the same for anything else.

**Comment by Sporkinator** [comment]
The bug might only exist in the OpenGL builds. I had somebody test it yesterday, and they could not reproduce it in DX12, but saw it in OpenGL. I can't run the DX12 build to test that, though.

**Comment by fgsfdsfgs** [comment]
Yeah well this repo only has the OpenGL backend. I *am* testing OpenGL builds.

**Comment by dgeneratioN** [comment]
just tested the nightly on w10 and still getting problems. It seems like there is a 
`if angle<0 then angle=0` when rotating clockwise.

**Comment by dgeneratioN** [comment]
So I did some testing, and it seems like when his `mario->faceAngle[1] > 32767` it goes to `-32768`
but not the other way around

**Comment by GateGuy** [comment]
Is someone working on a fix for this? If not, I can look into it.

**Comment by fgsfdsfgs** [comment]
I still cannot reproduce it.

**Comment by GateGuy** [comment]
For what it's worth, I still get this bug regardless of BETTERCAMERA and NODRAWDISTANCE settings. I haven't found anything about faceAngle[1] looping around, but I found what should set Mario's position when turning and it looks pretty straightforward (changing the 16.0f changes the rate at which Mario turns):

m->faceAngle[1] -= m->controller->stickX * 16.0f;
From act_holding_pole() and act_top_of_pole in mario_actions_automatic.c (this bug happens regardless of whether you're on the middle of a pole/tree or handstanding at the top)

**Comment by gblues** [comment]
I can reproduce it (Windows 10, Xbox 1 controller w/ wireless adapter, *not* using bluetooth).

The behavior I see:
- if pushing left on the stick, mario spins around as you expect until you release the stick
- if pushing right on the stick, the first time Mario will rotate until his back is to the camera and then stop (so the amount he turns varies depending on what angle he is to the camera)
- if push right a second time, Mario begins sliding down the tree and then will dismount once he hits the ground

**Comment by Madghostek** [comment]
Not sure if this is related, but on certain plugins on n64 emu the stick value can overflow when held fully to the right, does the issue happen when tilting joystick only a bit?

**Comment by GateGuy** [comment]
Yep, still happens.
